Tests/RegoTests/IREvaluatorTests.swift

```diff
@@ -211,13 +211,31 @@ struct IRStatementTests {
         // expectResult - if an explicit value is set, we will compare that to the resultSet.
         var expectResult: ResultSet? = nil
         var expectUndefined: Bool = false
-    }
 
-    func mergeLocals(_ lhs: Locals, _ rhs: Locals?) -> Locals {
-        guard let rhs else {
-            return lhs
+        // Convenience initializer that accepts dictionary-style locals
+        init(
+            description: String,
+            stmt: any Statement,
+            locals: [Int: Any] = [:],
+            funcs: [IR.Func] = [],
+            staticStrings: [String] = [],
+            expectLocals: [Int: Any]? = nil,
+            ignoreLocals: [Local] = [],
+            expectError: Bool = false,
+            expectResult: ResultSet? = nil,
+            expectUndefined: Bool = false
+        ) {
+            self.description = description
+            self.stmt = stmt
+            self.locals = Locals(from: locals)
+            self.funcs = funcs
+            self.staticStrings = staticStrings
+            self.expectLocals = expectLocals.map { Locals(from: $0) }
+            self.ignoreLocals = ignoreLocals
+            self.expectError = expectError
+            self.expectResult = expectResult
+            self.expectUndefined = expectUndefined
```

Comment:
I'd suggest removing this in favor of having Locals conform to [ExpressibleByDictionaryLiteral](https://developer.apple.com/documentation/swift/expressiblebydictionaryliteral):

```
extension Locals: ExpressibleByDictionaryLiteral {
    public typealias Key = Int
    public typealias Value = RegoValue

    public init(dictionaryLiteral elements: (Key, Value)...) {
        guard elements.count > 0 else {
            self = Locals()
            return
        }
        
        let maxLocal: Int = elements.reduce(into: 0) { max, elem in
            max = Swift.max(max, elem.0)
        }
        
        var result = Locals(repeating: nil, count: maxLocal + 1)
        
        for (idx, value) in elements {
            result[IR.Local(idx)] = value
        }
        self = result
    }
}
```